cranelift/codegen/src/machinst/mod.rs

```diff
@@ -0,0 +1,288 @@
+//! |        |
+//! |    VCode<arch_backend::Inst>   (machine instructions:
+//! |        |                        - vcode.final_block_order is filled in.
+//! |        |                        - new insn sequence from regalloc is
+//! |        |                          placed back into vcode and block
+//! |        |                          boundaries are updated.)
+//! |        | [redundant branch/block
+//! |        |  removal]
+//! |        |
+//! |    VCode<arch_backend::Inst>   (machine instructions:
+//! |        |                        - all blocks that were just an
+//! |        |                          unconditional branch are removed.)
+//! |        |
+//! |        | [branch finalization
+//! |        |  (fallthroughs)]
+//! |        |
+//! |    VCode<arch_backend::Inst>   (machine instructions:
+//! |        |                        - all branches are in lowered one-
+//! |        |                          target form, but targets are still
+//! |        |                          block indices.)
+//! |        |
+//! |        | [branch finalization
+//! |        |  (offsets)]
+//! |        |
+//! |    VCode<arch_backend::Inst>   (machine instructions:
+//! |        |                        - all branch offsets from start of
+//! |        |                          function are known, and all branches
+//! |        |                          have resolved-offset targets.)
+//! |        |
+//! |        | [MemArg finalization]
+//! |        |
+//! |    VCode<arch_backend::Inst>   (machine instructions:
+//! |        |                        - all MemArg references to the constant
+//! |        |                          pool are replaced with offsets.
+//! |        |                        - all constant-pool data is collected
+//! |        |                          in the VCode.)
+//! |        |
+//! |        | [binary emission]
+//! |        |
+//! |    Vec<u8>                     (machine code!)
+//! |
+
+#![allow(unused_imports)]
+
+use crate::binemit::{
+    CodeInfo, CodeOffset, CodeSink, MemoryCodeSink, RelocSink, StackmapSink, TrapSink,
+};
+use crate::entity::EntityRef;
+use crate::entity::SecondaryMap;
+use crate::ir::condcodes::IntCC;
+use crate::ir::ValueLocations;
+use crate::ir::{DataFlowGraph, Function, Inst, Opcode, Type, Value};
+use crate::isa::RegUnit;
+use crate::result::CodegenResult;
+use crate::settings::Flags;
+use crate::HashMap;
+use alloc::boxed::Box;
+use alloc::vec::Vec;
+use core::fmt::Debug;
+use core::iter::Sum;
+use regalloc::Map as RegallocMap;
+use regalloc::RegUsageCollector;
+use regalloc::{RealReg, RealRegUniverse, Reg, RegClass, SpillSlot, VirtualReg, Writable};
+use smallvec::SmallVec;
+use std::hash::Hash;
+use std::string::String;
+use target_lexicon::Triple;
+
+pub mod lower;
+pub use lower::*;
+pub mod vcode;
+pub use vcode::*;
+pub mod compile;
+pub use compile::*;
+pub mod blockorder;
+pub use blockorder::*;
+pub mod abi;
+pub use abi::*;
+pub mod pp;
+pub use pp::*;
```

Comment:
`pp` isn't something I can guess the meaning of without more context. What would you think about renaming it to `pretty-print.rs` or so?

Reply:
Renamed!